src/Dispatcher.js

```diff
@@ -11,17 +14,49 @@ export default class ReduxDispatcher {
     redux: PropTypes.object.isRequired
   };
 
-  constructor(props) {
-    this.dispatcher = createDispatcher();
-    this.dispatcher.receiveStores(props.stores);
+  getChildContext() {
+    return { redux: this };
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.dispatcher.receiveStores(nextProps.stores);
+  constructor(props, context) {
+    super(props, context);
+
+    this.subscriptions = [];
+    this.emitChange = this.emitChange.bind(this);
+    this.dispatch = this.dispatch.bind(this);
+
+    const initialAtom = dispatch(props.store, undefined, {});
+    this.setAtom(initialAtom);
   }
 
-  getChildContext() {
-    return { redux: this.dispatcher };
+  dispatch(action) {
+    const nextAtom = dispatch(this.props.store, this.atom, action);
+    this.setAtom(nextAtom);
+  }
+
+  setAtom(atom) {
+    this.atom = atom;
+    if (this.state) {
+      this.setState({ atom });
```

Comment:
I think these concepts are neither fancy nor complicated, but I understand your reasoning. I was once scared by those ideas, too; if avoiding the functional lingo means the library is more approachable to more people, then it's okay I guess. :+1: 

Too bad we have this FP-stigma, though. JS community could learn so much from FP (as it already did).